File: python/41d_1829B_BlankSpace.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = input().split()
        cnt, pos = 0, 0
        for i in range(n):
            if arr[i] == "0":
                cnt = max(cnt, i - pos + 1)
            else:
                pos = i + 1
        ans.append(cnt)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/41d_1829B_BlankSpace.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, temp, pos, cnt, j;
    std::vector<int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        pos = 0, cnt = 0, j = 0;
        for(j = 0; j < n; j++){
            std::cin>>temp;
            if (temp == 0){
                cnt = std::max(cnt, j - pos + 1);
            }
            else{
                pos = j + 1;
            }
        }
        ans.push_back(cnt);
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}